kubeadm/v1.15.0/KubeClusterHelper.psm1

```diff
@@ -0,0 +1,1360 @@
+    $existingMachinePath = [Environment]::GetEnvironmentVariable("Path",[System.EnvironmentVariableTarget]::Machine)
+    [Environment]::SetEnvironmentVariable("Path", $existingMachinePath + ";$env:ProgramFiles\Docker", [EnvironmentVariableTarget]::Machine)
+
+    $cmd = get-command docker.exe -ErrorAction SilentlyContinue
+    if (!$cmd)
+    {
+        $dockerVersion = $Version
+        DownloadFile  "https://master.dockerproject.org/windows/x86_64/${dockerVersion}.zip" -Destination "$env:TEMP\$dockerVersion.zip" 
+        Expand-Archive -Path "$env:TEMP\$dockerVersion.zip" -DestinationPath $env:ProgramFiles -Force
+        dockerd --register-service
+        Start-Service Docker -ErrorAction Stop
+        $Global:Configuration += @{
+            InstallDocker = $true;
+        }
+        WriteKubeClusterConfig
+    }
+}
+
+function InstallDockerImages()
+{
+    if (!(docker images $Global:NanoserverImage -q))
+    {
+        docker pull $Global:NanoserverImage
+        if (!$LastExitCode) {
+            throw "Failed to pull $Global:NanoserverImage"
+        }
+    }
+    docker tag $Global:NanoserverImage mcr.microsoft.com/windows/nanoserver:latest
+    if (!(docker images $Global:ServercoreImage -q))
+    {
+        docker pull $Global:ServercoreImage
+        if (!$LastExitCode) {
+            throw "Failed to pull $Global:ServercoreImage"
+        }
+    }
+    docker tag $Global:ServercoreImage mcr.microsoft.com/windows/servercore:latest
+}
+
+function InstallPauseImage()
+{
+    # Prepare POD infra Images
+    $infraPodImage=docker images $Global:PauseImage -q
+    if (!$infraPodImage)
+    {
+        Write-Host "No infrastructure container image found. Pulling $Global:PauseImage"
+        docker pull $Global:PauseImage
+        if ($LastExitCode) {
+            throw "Failed to pull $Global:PauseImage"
+        }
+    }
+}
+
+function InstallKubernetesBinaries()
+{
+    Param(
+    [parameter(Mandatory = $true)] $Source,
+    $DestinationPath
+    ) 
+
+    $existingPath = [Environment]::GetEnvironmentVariable("Path", [System.EnvironmentVariableTarget]::Machine)
+
+    # For current shell Path update
+    $env:path += ";$DestinationPath\kubernetes\node\bin"
+    # For Persistent across reboot
+    [Environment]::SetEnvironmentVariable("Path", $existingPath + ";$DestinationPath\kubernetes\node\bin", [EnvironmentVariableTarget]::Machine)
+
+    $env:KUBECONFIG = $(GetKubeConfig)
+    [Environment]::SetEnvironmentVariable("KUBECONFIG", $(GetKubeConfig), [EnvironmentVariableTarget]::Machine)
+
+    $Release = "1.15"
+    if ($Source.Release)
+    {
+        $Release = $Source.Release
+    }
+    $Url = "https://dl.k8s.io/v${Release}/kubernetes-node-windows-amd64.tar.gz"
+    if ($Source.Url)
+    {
+        $Url = $Source.Url
+    }
+    DownloadFile -Url  $Url -Destination $Global:BaseDir/kubernetes-node-windows-amd64.tar.gz
+    tar -zxvf $Global:BaseDir/kubernetes-node-windows-amd64.tar.gz -C $DestinationPath
```

Comment:
*open question:* should we be cleaning up the tarball so it doesn't take up space on the machine/device?

Reply:
This gets cleaned up when we delete C:/ProgramData/Kubernetes (for me anyway)